workbench-core/example/app/src/routes/datasetRoutes.ts

```diff
@@ -81,19 +92,57 @@ export function setUpDSRoutes(
   router.post(
     '/datasets/:datasetId/share',
     wrapAsync(async (req: Request, res: Response) => {
-      if (req.params.datasetId.match(uuidWithLowercasePrefixRegExp(dataSetPrefix)) === null) {
```

Comment:
Don't remove this. you are using `req.params.datasetId` later in your implementation (which is the right thing to do!), but you removed its validation!

Reply:
This is dead code, since authZ's static route config wont let a param that doesnt match get to the route handler